libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2175,6 +2175,14 @@ bool AP_Mission::get_next_cmd(uint16_t start_index, Mission_Command& cmd, bool i
                 return false;
             }
 
+            // handle jump item that points to itself
+            if ((temp_cmd.content.jump.target == cmd_index)) {
+                // This jump points to itself - move on to next mission item
+                // To-Do: log an error? 
+                cmd_index++;
```

Comment:
Thanks for this.
We could perhaps use a SEND_TEXT to alert the user as to why we're skipping the jump.

Reply:
I guess one thing that'd I'd check is that there's no change in behaviour if there are two jumps in a row.  the 1st one points to itself and the 2nd one points to the 1st one.

Reply:
> I guess one thing that'd I'd check is that there's no change in behaviour if there are two jumps in a row. the 1st one points to itself and the 2nd one points to the 1st one.

* If both waypoints are infinite jumps, it will skip both.
* If jump waypoint 1 is infinite and jump waypoint 2 is finite, then mission will result in "skip-then-jump" `n-jumps` (where `n-jumps` is jump waypoint 2 count).

Reply:
> I guess one thing that'd I'd check is that there's no change in behaviour if there are two jumps in a row. the 1st one points to itself and the 2nd one points to the 1st one.

To further clarify:

If there are two infinite jump loops in a row, both pointing to the first JUMP command in the sequence, it will still bail into RTL.

This is existing logic in ArduPilot that handles endless loops in this way.

The `max_loops` counter is decremented for each jump performed, once it reaches zero it bails and vehicle falls into RTL.

Reply:
> Thanks for this. We could perhaps use a SEND_TEXT to alert the user as to why we're skipping the jump.

I've can modify to send a message if the get_next_cmd is called with `send_gcs_msg == true`:
```c++
if (send_gcs_msg) {
    GCS_SEND_TEXT(MAV_SEVERITY_NOTICE, "Mission: Waypoint %u jumps to itself", cmd_index);
}
```

But that message is coming out more than once. Seems like it might be for each of the calls through `get_next_cmd` is sending out.

Reply:
The question is whether we want these self pointing waypoints to be considered bad when decrementing the max_loop jump count or not.

[decrementing `max_loops`](https://github.com/ArduPilot/ardupilot/pull/28604/files#diff-1500d3a42eeb51ad59732e97da1aa22b07cf37676664f4b5a0e21ea2f80cadd2R2168) before this check essentially halves the amount of jumps in this two infinite-jump-waypoint case (of course those jumps need to include the self-pointing jump waypoint).

Now that may be fine, but I can't say my mission plans have "advanced" jumps in them...

Reply:
as long as it's not getting stuck or crashing then I'm happy.  We probably should have a mission pre-arm check that checks for all kinds of potential issues including jump commands that jump to itself

Reply:
If the send-text is spamming then I'm ok removing it